docs/adding-new-cloud-provider.md

```diff
@@ -0,0 +1,50 @@
+# How to add a new Cloud Provider
+
+## Add provider specific code to the cloud credentials operator 
+The Cloud Credentials Operator has three modes
+
+- *Manual* is a recommended mode. In this mode a user manually performs the job of the cloud credential operator.  
+- *Mint* mode. In this mode the admin credential is stored in kube-system namespace and then used to create fine-grained permission for in-cluster components.
+- *Passthrough* mode. In this mode a single credential having all the required permissions is stored in kube-system namespace and passed to other in-cluster components. This credential cannot mint additional credentials.  
+
+We need a new provider to support at least the *Manual* mode. The other two modes are optional. Learn more about CCO modes [here](../README.md#modes-of-operation). 
+
+For adding a new provider we need new code at the following places
+
+1) Create a ProviderSpec API that holds the specifications for the credentials request. Refer aws implementation [here](../pkg/apis/cloudcredential/v1/types_aws.go).
+2) Create a ProviderStatus API that holds the status of the credentials request. Refer aws implementation [here](../pkg/apis/cloudcredential/v1/types_aws.go).
```

Comment:
Do we really require ProviderStatus if only Manual mode will be supported?

Reply:
Nope, moved it down